src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +280,839 @@ static DamageTable D_808792D4 = {
     /* Powder Keg     */ DMG_ENTRY(1, 0x0),
 };
 
-// sColChkInfoInit
-static CollisionCheckInfoInit D_808792F4 = { 3, 100, 100, 80 };
+static CollisionCheckInfoInit sColChkInfoInit = { 3, 100, 100, 80 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_808792FC[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_S8(hintId, 13, ICHAIN_CONTINUE),
     ICHAIN_F32_DIV1000(gravity, -1000, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 1400, ICHAIN_STOP),
 };
 
-#endif
+void EnDodongo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static s32 D_80879308[] = {
+        0x00000002, 0x00080000, 0xFFFFFFFF, 0xFFFFFF40, 0xFFFFFF00, 0xFFFFFF00, 0x08000000, 0x00000000, 0x00000000,
+    };
+
+    EnDodongo* this = THIS;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->unk_32C.r = 255;
+    this->unk_32C.g = 255;
+    this->unk_32C.a = 200;
+    this->unk_330.r = 255;
+    this->unk_330.g = 10;
+    this->unk_330.a = 200;
+    Math_Vec3f_Copy(&this->unk_314, &D_801C5DB0);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 48.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06008318, &D_06004C20, this->jointTable, this->morphTable, 31);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    Collider_InitAndSetJntSph(globalCtx, &this->collider2, &this->actor, &sJntSphInit2, this->collider2Elements);
+    Collider_InitAndSetJntSph(globalCtx, &this->collider1, &this->actor, &sJntSphInit1, this->collider1Elements);
+    Collider_InitAndSetJntSph(globalCtx, &this->collider3, &this->actor, &sJntSphInit3, this->collider3Elements);
+
+    for (i = 0; i < ARRAY_COUNT(this->collider2Elements); i++) {
+        this->collider2.elements[i].info.elemType = ELEMTYPE_UNK2;
+        this->collider2.elements[i].info.bumper.dmgFlags = 0x77C34FE6;
+    }
+
+    Effect_Add(globalCtx, &this->unk_338, 2, 0, 0, &D_80879308);
+    if (!this->actor.params) {
```

Comment:
```suggestion
    if (this->actor.params == 0) {
```
This isn't really a boolean, it just happens to have two values. My guess is it's the two different sizes of Dodongo.